// ts-jest/dist/experimental/option-validator.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.validateOptions = void 0;
const v = __importStar(require("valibot"));
/**
 * Valibot schema for ts-jest options validation
 */
const tsJestOptionsSchema = v.strictObject({
    tsconfig: v.optional(v.union([v.string(), v.record(v.string(), v.any())])),
    isolatedModules: v.optional(v.boolean()),
    compiler: v.optional(v.string()),
    astTransformers: v.optional(v.record(v.string(), v.any())),
    diagnostics: v.optional(v.union([v.boolean(), v.record(v.string(), v.any())])),
    babelConfig: v.optional(v.union([v.boolean(), v.string(), v.record(v.string(), v.any())])),
    stringifyContentPathRegex: v.optional(v.union([v.string(), v.instance(RegExp)])),
    useESM: v.optional(v.boolean()),
});
/**
 * @internal
 *
 * Validates `ts-jest` transformer options with both TypeScript types and runtime validation about unrecognized options to help catch typos and configuration errors.
 */
const validateOptions = (options) => {
    try {
        v.parse(tsJestOptionsSchema, options);
    }
    catch (error) {
        if (error instanceof v.ValiError) {
            throw new Error(error.message);
        }
        throw error;
    }
};
exports.validateOptions = validateOptions;
